cmd/alerts/silence/list_silence.go

```diff
@@ -33,25 +33,42 @@ type Silence struct {
 	StartsAt  string `json:"startsAt"`
 }
 
+type listSilenceCmd struct {
+	clusterID string
+	reason    string
+}
+
 func NewCmdListSilence() *cobra.Command {
-	return &cobra.Command{
+	listSilenceCmd := &listSilenceCmd{}
+	cmd := &cobra.Command{
 		Use:               "list <cluster-id>",
 		Short:             "List all silences",
 		Long:              `print the list of silences`,
 		Args:              cobra.ExactArgs(1),
 		DisableAutoGenTag: true,
 		Run: func(cmd *cobra.Command, args []string) {
-			ListSilence(args[0])
+			listSilenceCmd.clusterID = args[0]
+			ListSilence(listSilenceCmd)
 		},
 	}
+
+	cmd.Flags().StringVar(&listSilenceCmd.reason, "reason", "", "The reason for this command, which requires elevation, to be run (usualy an OHSS or PD ticket)")
+	_ = cmd.MarkFlagRequired("reason")
+
+	return cmd
 }
 
-func ListSilence(clusterID string) {
+func ListSilence(cmd *listSilenceCmd) {
```

Comment:
What do you think of just having that function take two parameters - a "reason" and a "clusterId"? This decouples it entirely from the cobra library and makes it a bit easier to test.

Reply:
Hmm, this is actually already decoupled. `listSilenceCmd` is just a struct and has nothing to do with cobra, maybe the name is causing confusion? But this is how we do it for the other commands as well (before this change).